data/meson.build

```diff
@@ -1,52 +1,47 @@
-install_data(
-    meson.project_name() + '.gschema.xml',
-    install_dir: join_paths(get_option('datadir'), 'glib-2.0', 'schemas')
-)
+datadir = get_option('prefix') / get_option('datadir')
+podir = meson.project_source_root() / 'po'
 
-i18n.merge_file (
+i18n.merge_file(
+    type: 'desktop',
     input: meson.project_name() + '.desktop.in',
-    output: meson.project_name() + '.desktop',
+    output: '@BASENAME@',
+    po_dir: podir / 'extra',
     install: true,
-    install_dir: join_paths(get_option('datadir'), 'applications'),
-    po_dir: join_paths(meson.source_root (), 'po', 'extra'),
-    type: 'desktop'
+    install_dir: datadir / 'applications'
 )
 
-i18n.merge_file (
+i18n.merge_file(
+    type: 'desktop',
     input: 'open-pantheon-terminal-here.desktop.in',
-    output: 'open-pantheon-terminal-here.desktop',
+    output: '@BASENAME@',
+    po_dir: podir / 'extra',
     install: true,
-    install_dir: join_paths(get_option('datadir'), 'applications'),
-    po_dir: join_paths(meson.source_root (), 'po', 'extra'),
-    type: 'desktop'
+    install_dir: datadir / 'applications'
 )
 
-i18n.merge_file (
+i18n.merge_file(
     input: meson.project_name() + '.appdata.xml.in',
-    output: meson.project_name() + '.appdata.xml',
+    output: '@BASENAME@',
+    po_dir: podir / 'extra',
     install: true,
-    install_dir: join_paths(get_option('datadir'), 'metainfo'),
-    po_dir: join_paths(meson.source_root (), 'po', 'extra')
+    install_dir: datadir / 'metainfo'
 )
 
-test (
-    'Validate desktop file',
-    find_program('desktop-file-validate'),
-    args: join_paths(meson.current_build_dir (), meson.project_name() + '.desktop')
-)
+install_data(meson.project_name() + '.gschema.xml', install_dir: datadir / 'glib-2.0' / 'schemas')
+install_data('pantheon_terminal_process_completion_notifications.fish', install_dir: datadir / 'fish' / 'vendor_conf.d')
+install_data('enable-zsh-completion-notifications', install_dir: datadir / meson.project_name())
```

Comment:
Can we keep these expanded on several lines as they were? I think this is a lot harder to read at a glance